File: apps/backend_go/internal/application/use_case/get_sync_status_usecase.go
package use_case

import (
	"context"
	"gold-vola-bunseki/backend/internal/application/port"
	"gold-vola-bunseki/backend/internal/domain"
)

/*
 * GetSyncStatusUseCase はデータ同期の状況を取得するユースケースです。
 * @responsibility: リポジトリからDBの最新状態を取得し、同期が正常であるかを判定する。
 */
type GetSyncStatusUseCase struct {
	sessionRepo port.SessionRepository
}

func NewGetSyncStatusUseCase(sessionRepo port.SessionRepository) *GetSyncStatusUseCase {
	return &GetSyncStatusUseCase{
		sessionRepo: sessionRepo,
	}
}

func (u *GetSyncStatusUseCase) Execute(ctx context.Context) (*domain.SyncStatus, error) {
	/* DBから最新の足、セッション、総件数を取得 */
	status, err := u.sessionRepo.GetSyncStatus(ctx)
	if err != nil {
		return nil, err
	}
	return status, nil
}

File: apps/backend/src/application/use_case/getSyncStatusUseCase.ts
import { SyncRepositoryPort } from '../port/syncRepositoryPort';
import { SyncStatus } from '../../domain/entities/syncStatus';

/**
 * GetSyncStatusUseCase は現在の同期状況を取得するユースケースです。
 * @responsibility: 同期ステータスを組み立て、コントローラーに提供する。
 */
export class GetSyncStatusUseCase {
  constructor(private syncRepo: SyncRepositoryPort) {}

  /**
   * execute は同期ステータスを取得し、必要があればビジネスルールに基づく補正を行います。
   * @return: SyncStatus
   */
  async execute(): Promise<SyncStatus> {
    const status = await this.syncRepo.getSyncStatus();
    // ここで複雑な仕様（例えば、特定の時間帯ならStale判定を変える等）があれば将来追加できます。
    return status;
  }
}
